src/dataset.py

```diff
@@ -98,43 +69,33 @@ def _augment_with_salary(record: Dict[str, str]) -> Dict[str, str]:
     return context
 
 
-def badge_lookup(badge: str) -> str:
-    try:
-        r = BADGE_DATASET.get(badge)
-        if not r:
-            html = "<p><b>No officer found for this badge number</b></p>"
-        else:
-            context = _augment_with_salary(r)
-            html = render_template("officer.j2", **context)
-
-    except Exception as err:
-        print(f"Error: {err}")
-        html = f"<p><b>Error:</b> {err}"
-
-    print("Final HTML:\n{}".format(html))
-    return html
-
-
-def _sort_names(record: _DataDict):
-    for r in sorted(record.values(), key=lambda x: x[C.FIRST]):
+def _sort_names(record: List[Tuple]) -> Tuple:
```

Comment:
I might suggest moving this function up before the augment, and instead converting the record at this point to a `NamedTuple`. This would make access to the return values a little bit clearer (also note I've changed the variable name and the return type):

```python
class RosterRecord(NamedTuple):
    serial: int
    first: str
    last: str
    middle: str
    title: str
    unit: str

# Usage
def _sort_names(records: List[Tuple]) -> Generator[RosterRecord, None, None]: 
    # Convert to named tuples
    records = [RosterRecord(*r) for r in records]
    for r in sorted(records, key=lambda x: x.last): 
        yield r


# Later usage
record.title
record.unit
# Etc
```